x-pack/plugins/gis/public/elasticsearch_geo_utils.js

```diff
@@ -39,44 +36,68 @@ export function hitsToGeoJson(hits, geoFieldName, geoFieldType) {
         }
       }
 
-      return {
-        type: 'Feature',
-        geometry: geometry,
-        properties: properties
-      };
+      return geometries.map(geometry => {
+        return {
+          type: 'Feature',
+          geometry: geometry,
+          properties: properties
+        };
+      });
     });
 
   return {
     type: 'FeatureCollection',
-    features: features
+    features: _.flatten(features)
+  };
+}
+
+function pointGeometryFactory(lat, lon) {
+  return {
+    type: 'Point',
+    coordinates: [lon, lat]
   };
 }
 
 export function geoPointToGeometry(value) {
-  let lat;
-  let lon;
+  if (!value) {
+    return [];
+  }
+
   if (typeof value === 'string') {
     const commaSplit = value.split(',');
     if (commaSplit.length === 1) {
       // Geo-point expressed as a geohash.
       throw new Error(`Unable to convert to geojson, geohash not supported`);
     }
     // Geo-point expressed as a string with the format: "lat,lon".
-    lat = parseFloat(commaSplit[0]);
-    lon = parseFloat(commaSplit[1]);
-  } else if (Array.isArray(value)) {
-    // Geo-point expressed as an array with the format: [ lon, lat]
-    lat = value[1];
-    lon = value[0];
-  } else if (value !== null && typeof value === 'object') {
-    lat = value.lat;
-    lon = value.lon;
+    const lat = parseFloat(commaSplit[0]);
+    const lon = parseFloat(commaSplit[1]);
+    return [pointGeometryFactory(lat, lon)];
   }
 
-  return {
-    type: 'Point',
-    coordinates: [lon, lat]
-  };
+  if (typeof value === 'object' && _.has(value, 'lat') && _.has(value, 'lon')) {
+    // Geo-point expressed as an object with the format: { lon, lat }
+    return [pointGeometryFactory(value.lat, value.lon)];
+  }
+
+  if (!Array.isArray(value)) {
+    throw new Error(`Unsupported geo_point value: ${value}`);
+  }
+
+  if (value.length === 2
+      && typeof value[0] === 'number'
+      && typeof value[1] === 'number') {
+    // Geo-point expressed as an array with the format: [lon, lat]
+    const lat = value[1];
+    const lon = value[0];
+    return [pointGeometryFactory(lat, lon)];
+  }
+
+  // Geo-point expressed as an array of values
+  const points = value.map(itemInValueArray => {
```

Comment:
This could be done in a single [hidden] loop with `reduce` instead of `map` -> `flatten`:

```js
return value.reduce((itemInValueArray, points) => points.concat(geoPointToGeometry(itemInValueArray)), []);
```